src/main/java/com/direwolf20/buildinggadgets/common/building/modes/ExchangingVerticalMode.java

```diff
@@ -0,0 +1,36 @@
+package com.direwolf20.buildinggadgets.common.building.modes;
+
+import com.direwolf20.buildinggadgets.common.BuildingGadgets;
+import com.direwolf20.buildinggadgets.common.building.IPlacementSequence;
+import com.direwolf20.buildinggadgets.common.building.IValidatorFactory;
+import com.direwolf20.buildinggadgets.common.building.placement.Column;
+import com.direwolf20.buildinggadgets.common.building.placement.Single;
+import com.direwolf20.buildinggadgets.common.tools.GadgetUtils;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+
+public class ExchangingVerticalMode extends AbstractMode {
+
+    private static final ResourceLocation NAME = new ResourceLocation(BuildingGadgets.MODID, "vertical_column");
+
+    public ExchangingVerticalMode(IValidatorFactory validatorFactory) {
+        super(validatorFactory);
+    }
+
+    @Override
+    public IPlacementSequence computeCoordinates(EntityPlayer player, BlockPos hit, EnumFacing sideHit, ItemStack tool) {
+        int range = GadgetUtils.getToolRange(tool);
+        if (sideHit.getAxis().isVertical())
+            return Single.create(hit);
+        return Column.centerAt(hit, EnumFacing.Axis.Y, range);
```

Comment:
```suggestion
        return Column.centerAt(hit, (sideHit.getAxis().isVertical() ? player.getHorizontalFacing().getAxis() : Axis.Y), range);
```
Replace lines 26-28 with this one.